### PR TITLE
braintree - add customGateway config and export environment class

### PR DESCRIPTION
**Pull request by SharonGrossman**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/braintree/braintree_node/blob/master/lib/braintree/environment.js#L3
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## braintree ([<kbd>unpkg</kbd>](https://unpkg.com/browse/braintree@latest/))
was missing the following properties:
1. GraphQL
1. version
1. errorTypes
1. exceptions
1. PaymentInstrumentTypes

<details>
<summary>as well as these 4 other properties...</summary>
<p>WebhookNotification, TestingGateway, UsBankAccountVerification, ValidationErrorCodes</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 1cb092b1e91b5c3ecacfc7a474497b5294b01fef
</p>

File: types/braintree/tslint.json [hunk 1]
```diff
@@ -2,6 +2,8 @@
     "extends": "@definitelytyped/dtslint/dt.json",
     "rules": {
         "no-empty-interface": false,
-        "strict-export-declare-modifiers": false
+        "no-unnecessary-class": false,
```
Comment (sheetalkamat) [inline types/braintree/tslint.json:5]: Do not add these extra rules.. instead fix the errors.

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/braintree/tslint.json (hunk 1 above)
Comment (SharonGrossman) [reply types/braintree/tslint.json]: not sure about the whitespace error though, I've seen a lot of packages add this rule, and it seems lint / format is removing the whitespace and then tslint is angry for not having them - maybe I'm missing something

**Comment by SharonGrossman** [comment]
Would like another review please 🙏🏻

File: types/braintree/tslint.json (hunk 1 above)
Comment (SharonGrossman) [reply types/braintree/tslint.json]: Would like another review please  🙏🏻

**Review by RyanCavanaugh — APPROVED** [review]

**Comment by SharonGrossman** [comment]
Ready to merge

**Comment by simllll** [comment]
@SharonGrossman this PR breaks the types, as you renamed "Environment" to "EnvironmentType".. 
Environemnt is actually a "predefiend" set of configs.. see Environment.js of the braintree package.
image

Now refering it as EnvironmentType is not resolving the pre defined config during runtime (build works, but runtime fails).

Using Environment, would need extra definitinos for the Sandbox and Production,.. e.g. by adding:
```
static readonly Sandbox: Environment;
static readonly Production: Environment;

```
` environment: EnvironmentType;` is as config param therefore actually not supported, or at least I cannot find a way to get this running, and accoridng to the docs it's also not possible.

I've created a PR to fix it over here https://github.com/DefinitelyTyped/DefinitelyTyped/pull/64382

**Pull request merged**